### PR TITLE
make prefix variable optional

### PR DESCRIPTION
**Pull request by withersj888**
## Describe your changes

When providing a name for the Log Analytics Workspace (var.cluster_log_analytics_workspace_name) and a DNS prefix is not required for the cluster (dns_prefix argument within azurerm_kubernetes_cluster.main), the prefix variable is not required and should therefore be made optional.

## Issue number

#308 

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by withersj888** [comment]
@microsoft-github-policy-service agree

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by lonegunmanb** [comment]
Hi @withersj888 thanks for opening this pr, since it looks like a non-breaking change I'd like to defer it to 7.1.0 so we can verify that this pr won't introduce any breaking or surprise change to this module. Our next release is 7.0.0 and I'd like to put all breaking changes in this release, we'll skip the test for breaking changes, that's why we'd like to defer this pr.

Again, thanks for your contribution, I'll deal with your proposal right after we've released the 7.0.

**Comment by withersj888** [comment]
Hi @lonegunmanb, no worries, glad i could contribute.
I see the prepr-check has failed, but i'm not sure how to fix it - I ran make precommit before the PR.

**Comment by lonegunmanb** [comment]
@withersj888 This pr made `var.prefix` an optional variable, and our pipeline is enforcing a [sorting rule](https://github.com/Azure/terraform-azure-modules/blob/main/codex/logic_code/variables.tf.md#order-to-define-variable), so we need move this `var.prefix` to the right place.

I'm working on an [autofix](https://github.com/lonegunmanb/azure-verified-module-fix) tool that could sort tf code after a pr has been merged so we won't be bothered to take care of such things any more, but for now we need sort code manually.

**Comment by zioproto** [comment]
@withersj888 look at this section of the README:
https://github.com/Azure/terraform-azurerm-aks#pre-commit--pr-check--test

you need to start with the `pre-commit`:
```
$ docker run --rm -v $(pwd):/src -w /src mcr.microsoft.com/azterraform:latest make pre-commit
```

File: variables.tf [hunk 1]
```diff
@@ -1013,6 +1009,11 @@ variable "rbac_aad_tenant_id" {
   default     = null
 }
 
+variable "resource_group_name" {
```
Comment (lonegunmanb) [inline variables.tf:1012]: `resource_group_name` is still a required variable so we should move it back to the original place.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Hi @withersj888 thanks for the update, we might need to move `var.resource_group_name` back to pass the `pr-check`.

**Comment by withersj888** [comment]
Ah, OK. I realise now that the variables are ordered by required and optional and then alphabetical.

________________________________

**Comment by lonegunmanb** [comment]
I'm closing this pr in favor of #382 , thanks for your contribution @withersj888 !

**Pull request closed without merging**